hw/battery/src/battery.c

```diff
@@ -818,7 +823,7 @@ battery_init(struct os_dev *dev, void *arg)
     }
     os_mutex_release(&battery_manager.bm_lock);
 
-    assert(i < BATTERY_MAX_COUNT);
+    assert(i <= BATTERY_MAX_COUNT);
```

Comment:
This change is not needed. Condition was correct.
On the other hand mutex release should be done later and that could be fixed.

Reply:
Agree! Thanks for the feedback! I didn't think that one through